protected/views/layouts/new_datasetpage.php

```diff
@@ -6,7 +6,7 @@
     <meta name="language" content="en" />
     <meta name="robots" content="noindex">
     <meta name="googlebot" content="noindex">
-    <meta name="citation_doi" content="<?php echo $mainSection->getReleaseDetails()['full_doi']; ?>">
+    <meta name="citation_doi" content="<?php echo CHtml::encode($this->fullDOI); ?>">
```

Comment:
Hi @rija 

I followed `pageTitle` in `view.php` to return `full_doi` in meta tag in `new_datasetpage.php`. However, I got this error:
```
Error 500
Property "DatasetController.fullDOI" is not defined.
```

Reply:
Hi @rija 

I tried the following replacement in `/view.php`:
```
$this->pageTitle = $mainSection->getReleaseDetails()['full_doi'];
```

And call `$this->pageTitle` inside meta tag in `/new_datasetpage.php`
```
<meta name="citation_doi" content="<?php echo CHtml::encode($this->pageTitle); ?>" >
```
The `full doi` can be shown in the web page properly.
![image](https://user-images.githubusercontent.com/64770635/100317240-fe4f4400-2ff6-11eb-94a5-d70468c91788.png)

So, I think the problem is `$this->fullDOI` has not been properly defined. Do you know where should I look to about how to properly define a variable in view class?

Reply:
Hi @kencho51,

Regarding figuring out how ``pageTitle`` works.

Looking at the top of the ``DatasetController.php`` file, you have: 


```
class DatasetController extends Controller
```

If you suspect a class is doing something but you see no evidence of it in the code, check if it extends another class, in this case ``Controller.php``.

If the code in that class still doesn't explain things, look at its header:

```
class Controller extends CController
```

and drill down to its parent class (a.k.a its superclass)


If you're wondering how the thing you see there works, have a look at this documentation:


https://www.yiiframework.com/doc/guide/1.1/en/basics.component#defining-and-using-a-component-property

That documentation also tells you what you need to do to get your code working

Reply:
Hi @rija 

I added `fullDOI` function to `CController.php` and it is working properly now. And the `behat test` is also passing.

Reply:
Hi @rija 
After looking at the `CController.php`, I tried to add `getFullDOI` and `setFullDOI` methods with reference to the `pageTitle`, this has been working and the `meta tag` was found inside `<head>` in HTML page.
But the `CController.php` is inside the `vendor` directory, so changing it is not a proper way?

Reply:
@kencho51 I'm not entirely sure what you need to do here but if you want to retrieve or change the value of `pageTitle` then this can be done by calling the `setPageTitle()` and `getPageTitle()` functions in [`Controller`](https://www.yiiframework.com/doc/api/1.1/CController) from within the `DatasetController` class. As @rija says, this is because the `DatasetController` class extends `Controller` class which itself extends `CController` Therefore, `CController`shares its functions with`DatasetController` so this class can use them.

Reply:
@kencho51 Looking at how to create the `<meta>` tag for `name="citation_doi"`, dataset web pages use the `new_column2` layout as defined in `actionView($id)` function in `DatasetController`. The `new_column2` layout itself uses `new_datasetpage` for rendering the `<head>` section and is provided with an array called `metaData` containing data it needs to use. You can add more items into the `metaData` array in `DatasetController` and then this can be accessed in `new_datasetpage.php`.